File: my_py_pkg/my_py_pkg/hw_status_publisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

from my_robot_interfaces.msg import HardwareStatus

class HardwarestatusPublisherNode(Node):
    def __init__(self):
        super().__init__("hardware_status_publisher")
        self.hw_status_publisher = self.create_publisher(HardwareStatus, "hardware_status", 10)
        self.timer_ = self.create_timer(1.0, self.publish_hw_status)
        self.get_logger().info("HW Publisher has been started.")

    def publish_hw_status(self):
        msg = HardwareStatus()
        msg.temperature = 1
        msg.are_motors_ready = True
        msg.debug_message = "Nothing special here"

        self.hw_status_publisher.publish(msg)

def main(args=None):
    rclpy.init(args=args)
    node = HardwarestatusPublisherNode()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: my_cpp_pkg/src/hw_status_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "my_robot_interfaces/msg/hardware_status.hpp"

class HardwarestatusPublisherNode: public rclcpp::Node {
public:
    HardwarestatusPublisherNode(): Node("hardware_status_publisher"){
        pub_ = this->create_publisher<my_robot_interfaces::msg::HardwareStatus>(
            "hardware_status", 10);
        timer_ = this->create_wall_timer(
            std::chrono::seconds(1),
            std::bind(&HardwarestatusPublisherNode::publishHardwareStatus, this));
        RCLCPP_INFO(get_logger(), "HW publisher has been started.");
    }
private:
    void publishHardwareStatus(){
        auto msg = my_robot_interfaces::msg::HardwareStatus();
        msg.temperature = 50;
        msg.are_motors_ready = false;
        msg.debug_message = "Motors are too hot.";
        pub_->publish(msg);
    }

    rclcpp::Publisher<my_robot_interfaces::msg::HardwareStatus>::SharedPtr pub_;
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char** argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<HardwarestatusPublisherNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}